### PR TITLE
Bug/pn schema compat fix

### PR DESCRIPTION
**Pull request by WilliamBerryiii**
Closes issue #1469 through the following changes:

The following changes to the Schema Generator were made in this PR: https://github.com/WilliamBerryiii/opcpublisherschemavalidator/pull/5 
- [X] Update the Schema Generator tool to produce a schema that only performs type validation on `ID` and `ExpandedNodeId` fields, removing value validation.
- [X] Update the Schema Generator tool to produce a schema that allows both `ID` and `ExpandedNodeId` to be present in an `OpcNodes` array item of the primary (current subschema)(Note: will not be attached to this repo; but we'll link back for thread consistency).
- [X] Update the Schema Generator tool's unit tests to cover the pn.json sample provided in PR #1464 (Note: will not be attached to this repo; but we'll link back for thread consistency).

In this Repository the commits below cover the following changes: 
- [X] Add updated backwards compatible schema to OPC Publisher project; rename prior schema as a "strict" schema.
- [X] Add unit tests to OPC UA Schema Validation tests in the OPC Publisher project to cover the pn.json sample provided in PR #1464 
- [X] Update schema spec doc included in the OPC Publisher project to include references to this conversation and adjustments made to the schema

@cristipogacean - if you'd like me to raise an issue to track adding additional unit tests to the core pn.json serialization, along the lines of the ones I added to the Schema Validator tests, I'd be happy to contribute some work there.

### CODE REVIEW

**Comment by WilliamBerryiii** [comment]
@karok2m - let me know if there's anything I need to do WRT this PR. Thanks for the main merge.

**Comment by cristipogacean** [comment]
@WilliamBerryiii, we did run in a strange issue recently with the schema validator. When using a `pn.json` file that has an UTF8 BOM header, the json parsing fails with the exception below. When the schema validation is disabled, everything is fine. You can very easily replicate this by applying the BOM to the file(s) we use in unit tests.

Nice was that running into this issue we found another really nasty bug on exception handling that caused loading the "broken" pn.json in an infinite loop. Be aware that this issue is not fixed yet on master - an easy fix though.

Would you have some capacity to look into this issue?

```
[00:34:12 INF Root] File Engine/pn_assets.json has changed, last known hash , new hash 4CDCFBD4D4FAF4ED9E26981542196B37C01B05062BAC879E7A86CC2EAF00A5B5, reloading... 
Exception thrown: 'System.Text.Json.JsonReaderException' in System.Text.Json.dll
Exception thrown: 'System.Text.Json.JsonReaderException' in System.Text.Json.dll
Exception thrown: 'System.Text.Json.JsonReaderException' in System.Text.Json.dll
[00:34:14 ERR Root] Error while reloading Engine/pn_assets.json 
System.Text.Json.JsonReaderException: '0xEF' is an invalid start of a value. LineNumber: 0 | BytePositionInLine: 0.
   at System.Text.Json.ThrowHelper.ThrowJsonReaderException(Utf8JsonReader& json, ExceptionResource resource, Byte nextByte, ReadOnlySpan`1 bytes)
   at System.Text.Json.Utf8JsonReader.ConsumeValue(Byte marker)
   at System.Text.Json.Utf8JsonReader.ReadFirstToken(Byte first)
   at System.Text.Json.Utf8JsonReader.ReadSingleSegment()
   at System.Text.Json.Utf8JsonReader.Read()
   at System.Text.Json.JsonDocument.Parse(ReadOnlySpan`1 utf8JsonSpan, Utf8JsonReader reader, MetadataDb& database, StackRowStack& stack)
   at System.Text.Json.JsonDocument.Parse(ReadOnlyMemory`1 utf8Json, JsonReaderOptions readerOptions, Byte[] extraRentedBytes)
   at System.Text.Json.JsonDocument.Parse(ReadOnlyMemory`1 json, JsonDocumentOptions options)
   at System.Text.Json.JsonDocument.Parse(String json, JsonDocumentOptions options)
   at Microsoft.Azure.IIoT.Validators.JsonSchemaDotNetSchemaValidator.Validate(Byte[] jsonBuffer, TextReader schemaReader) in C:\src\Industrial-IoT\common\src\Microsoft.Azure.IIoT.Validators.JsonSchemaDotNet\src\Default\JsonSchemaDotNetSchemaValidator.cs:line 43
   at Microsoft.Azure.IIoT.Serializers.NewtonSoft.NewtonSoftJsonSerializer.Deserialize(ReadOnlyMemory`1 buffer, Type type, TextReader schemaReader) in C:\src\Industrial-IoT\common\src\Microsoft.Azure.IIoT.Serializers.NewtonSoft\src\Default\NewtonSoftJsonSerializer.cs:line 91
   at Microsoft.Azure.IIoT.Serializers.SerializerEx.Deserialize(ISerializer serializer, String str, Type type, TextReader schemaReader) in C:\src\Industrial-IoT\common\src\Microsoft.Azure.IIoT.Abstractions\src\Serializers\Extensions\SerializerEx.cs:line 145
   at Microsoft.Azure.IIoT.Serializers.SerializerEx.Deserialize[T](ISerializer serializer, String json, TextReader schemaReader) in C:\src\Industrial-IoT\common\src\Microsoft.Azure.IIoT.Abstractions\src\Serializers\Extensions\SerializerEx.cs:line 199
   at Microsoft.Azure.IIoT.OpcUa.Edge.Publisher.Models.PublishedNodesJobConverter.Read(String publishedNodesContent, TextReader publishedNodesSchemaFile) in C:\src\Industrial-IoT\components\opc-ua\src\Microsoft.Azure.IIoT.OpcUa.Edge.Publisher\src\Storage\PublishedNodesJobConverter.cs:line 56
   at Microsoft.Azure.IIoT.OpcUa.Edge.Publisher.Engine.LegacyJobOrchestrator.RefreshJobFromFile() in C:\src\Industrial-IoT\components\opc-ua\src\Microsoft.Azure.IIoT.OpcUa.Edge.Publisher\src\Engine\LegacyJobOrchestrator.cs:line 274
Exception thrown: 'Xunit.Sdk.NotNullException' in xunit.assert.dll
```

**Comment by WilliamBerryiii** [comment]
@cristipogacean - yes, @suneetnangia or I will take a look!

**Comment by cristipogacean** [comment]
@WilliamBerryiii, @suneetnangia, do you have some time to look into the above mentioned topic? we would like to merge this PR soon

**Comment by suneetnangia** [comment]
> @WilliamBerryiii, @suneetnangia, do you have some time to look into the above mentioned topic? we would like to merge this PR soon

I'll have a look at it tomorrow, thanks for reminding Cristian.

**Comment by WilliamBerryiii** [comment]
@cristipogacean - squash and merged @suneetnangia's fix. Build/Test should be finished soon.

**Review by cristipogacean — APPROVED** [review]
:shipit:

File: modules/src/Microsoft.Azure.IIoT.Modules.OpcUa.Publisher/src/Microsoft.Azure.IIoT.Modules.OpcUa.Publisher.csproj [hunk 1]
```diff
@@ -39,8 +39,8 @@
     <None Update="publishednodes.json">
       <CopyToOutputDirectory>Always</CopyToOutputDirectory>
     </None>
-    <None Update="schemas\publishednodesschema.json">
-      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
+    <None Update="Schemas\strict-publishednodesschema.json">
```
Comment (karok2m) [inline modules/src/Microsoft.Azure.IIoT.Modules.OpcUa.Publisher/src/Microsoft.Azure.IIoT.Modules.OpcUa.Publisher.csproj:42]: Shouldn't we copy both of the schemas and not only the strict one ?

**Review by karok2m — APPROVED** [review]

File: modules/src/Microsoft.Azure.IIoT.Modules.OpcUa.Publisher/src/Microsoft.Azure.IIoT.Modules.OpcUa.Publisher.csproj (hunk 1 above)
Comment (karok2m) [reply modules/src/Microsoft.Azure.IIoT.Modules.OpcUa.Publisher/src/Microsoft.Azure.IIoT.Modules.OpcUa.Publisher.csproj:42]: Addressing this comment in #1565

**Pull request merged**